numfocus-sdg-2023-r1.md

```diff
@@ -0,0 +1,133 @@
+
+Visualizing astronomical objects together with their path and motion is a key
+feature of any Astrodynamics library, including poliastro.
+
+At the moment, poliastro supports two backends built on top of the Matplotlib
+library and the Plotly library. However, these libraries are not conceived for
+three-dimensional visualization and become a limitation when requiring complex
+visualization techniques like using textures.
+
+A solution proposed during the SciPy 2022 conference was to create a new backend
+based on PyVista. This library provides a Pythonic high-level API to The
+Visualization Toolkit (VTK) library.
+
+Another desired feature in poliastro is animating plots. This feature becomes
+essential when simulating the evolution of a system. With animated plots, users
+would be able to have better understanding on how the system evolves over time.
+
+This last feature is supported by Matplotlib, Plotly and the new proposed
+PyVista backends.
+
+
+7 - Benefit to project and community
+-------------------------------------
+
+The coding sprints that took place at SciPy 2022 conference turned out to be an
+excellent place to discuss with PyVista maintainers about using their library as
+a new plotting backend for poliastro.
+ 
+This proposal can generate the following benefits to the project and the
+community:
+
+* Users can generate and customize powerful and accurate visualizations and
+  animations of their simulations using poliastro. This would bring poliastro a
+  new 
+
+* poliastro would still remain as a pure Python, as PyVista is a high-level
+  Python wrapper of VTK. No need for maintainers and contributors to learn other
+  programming languages or complex graphics APIs. Everything is implemented
+  using Python.
+
+* Generic features can be implemented upstreamed to PyVista so their users can
+  benefit without requiring to install poliastro.
+  
+* Since PyVista is an affiliated NumFOCUS project, all the community can benefit
+  from the grant in one way or another.
+
+* The testing of the new backend can be performed using the pytest-pyvista
+  plugin for pytest. The adoption of this plugin in poliastro can generate
+  useful feedback for enhancing the plugin.
+
+All the proposed features would allow poliastro to be closer to commercial
+software while still remaining open-source and a reference project in the Python
+scientific community.
+
+
+8 - Amount requested
+--------------------
+
+Total amount requested: 4500$
+
+
+9 - Budget justification
+------------------------
+
+The project, as depicted in the "Timeline" section of this proposal, is expected
+to be completed within the course of 5 months. Starting date would be May 2023
+and the project would be completed by the end of September 2023.
+
+The grant is used as hourly rates and number of work hours for contractors. The
+requested budget is justified as follows:
+
+4500$ = 5 [month] * 15 [day / month] *  3 * [hour / day] * 20.00 [$ / hour] 
```

Comment:
So, more or less 12 hours a week, right?

Reply:
That's it. I think this is a reasonable amount of hours that I can accomplish over the course of 5 months.